corehq/apps/users/tasks.py

```diff
@@ -347,6 +348,3 @@ def process_reporting_metadata_staging():
                 user.save(fire_signals=False)
 
             record.delete()
-
-    if UserReportingMetadataStaging.objects.exists():
-        process_reporting_metadata_staging.delay()
```

Comment:
without this it will only process 100 records

Reply:
Ha, right. Reverted that change. Hopefully deleting the record when needed addresses this issue.